From 71547ab6782fd9b8c32f2bd6c6284b441fe6254f Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 4 Nov 2021 21:19:46 +0100
Subject: [PATCH] Add support for compressed kernel modules

compressed kernel modules may have a .ko.zstd suffix. Add support for
this in mkinitfs

also include /sbin/modprobe in the base
---
 features.d/ata.modules        |  2 +-
 features.d/base.files         |  1 +
 features.d/base.modules       |  2 +-
 features.d/cryptsetup.modules |  2 +-
 features.d/dhcp.modules       |  2 +-
 features.d/floppy.modules     |  2 +-
 features.d/kms.modules        |  4 ++--
 features.d/lvm.modules        | 10 +++++-----
 features.d/nbd.modules        |  2 +-
 features.d/network.modules    |  6 +++---
 features.d/usb.modules        |  6 +++---
 features.d/xenpci.modules     |  2 +-
 12 files changed, 21 insertions(+), 20 deletions(-)

diff --git a/features.d/ata.modules b/features.d/ata.modules
index 2a68a98..0213bc5 100644
--- a/features.d/ata.modules
+++ b/features.d/ata.modules
@@ -1 +1 @@
-kernel/drivers/ata/*.ko
+kernel/drivers/ata/*.ko*
diff --git a/features.d/base.files b/features.d/base.files
index 23718cb..c05684a 100644
--- a/features.d/base.files
+++ b/features.d/base.files
@@ -1,5 +1,6 @@
 /bin/busybox
 /bin/sh
+/sbin/modprobe
 /lib/mdev
 /sbin/apk
 /etc/modprobe.d/*.conf
diff --git a/features.d/base.modules b/features.d/base.modules
index f8f271d..ec440f1 100644
--- a/features.d/base.modules
+++ b/features.d/base.modules
@@ -1,2 +1,2 @@
-kernel/drivers/block/loop.ko
+kernel/drivers/block/loop.ko*
 kernel/fs/overlayfs
diff --git a/features.d/cryptsetup.modules b/features.d/cryptsetup.modules
index 1469be7..696687e 100644
--- a/features.d/cryptsetup.modules
+++ b/features.d/cryptsetup.modules
@@ -1,3 +1,3 @@
 kernel/crypto/*
 kernel/arch/*/crypto/*
-kernel/drivers/md/dm-crypt.ko
+kernel/drivers/md/dm-crypt.ko*
diff --git a/features.d/dhcp.modules b/features.d/dhcp.modules
index f9799e0..db067da 100644
--- a/features.d/dhcp.modules
+++ b/features.d/dhcp.modules
@@ -1 +1 @@
-kernel/net/packet/af_packet.ko
+kernel/net/packet/af_packet.ko*
diff --git a/features.d/floppy.modules b/features.d/floppy.modules
index 5677a9a..bd5df06 100644
--- a/features.d/floppy.modules
+++ b/features.d/floppy.modules
@@ -1 +1 @@
-kernel/drivers/block/floppy.ko
+kernel/drivers/block/floppy.ko*
diff --git a/features.d/kms.modules b/features.d/kms.modules
index b62e75b..2b6035b 100644
--- a/features.d/kms.modules
+++ b/features.d/kms.modules
@@ -2,5 +2,5 @@ kernel/drivers/char/agp
 kernel/drivers/gpu
 kernel/drivers/i2c
 kernel/drivers/video
-kernel/arch/x86/video/fbdev.ko
-kernel/drivers/gpu/drm/vc4/vc4.ko
+kernel/arch/x86/video/fbdev.ko*
+kernel/drivers/gpu/drm/vc4/vc4.ko*
diff --git a/features.d/lvm.modules b/features.d/lvm.modules
index 0437f5f..300c05d 100644
--- a/features.d/lvm.modules
+++ b/features.d/lvm.modules
@@ -1,5 +1,5 @@
-kernel/drivers/md/dm-mod.ko
-kernel/drivers/md/dm-snapshot.ko
-kernel/drivers/md/dm-cache.ko
-kernel/drivers/md/dm-raid.ko
-kernel/drivers/md/dm-thin-pool.ko
+kernel/drivers/md/dm-mod.ko*
+kernel/drivers/md/dm-snapshot.ko*
+kernel/drivers/md/dm-cache.ko*
+kernel/drivers/md/dm-raid.ko*
+kernel/drivers/md/dm-thin-pool.ko*
diff --git a/features.d/nbd.modules b/features.d/nbd.modules
index 0b0ca75..7e0b307 100644
--- a/features.d/nbd.modules
+++ b/features.d/nbd.modules
@@ -1 +1 @@
-kernel/drivers/block/nbd.ko
+kernel/drivers/block/nbd.ko*
diff --git a/features.d/network.modules b/features.d/network.modules
index 844e421..056576a 100644
--- a/features.d/network.modules
+++ b/features.d/network.modules
@@ -1,7 +1,7 @@
 kernel/drivers/net/ethernet
 kernel/drivers/net/phy
-kernel/net/packet/af_packet.ko
+kernel/net/packet/af_packet.ko*
 kernel/drivers/net/hyperv
 kernel/drivers/net/vmxnet3
-kernel/drivers/virtio/virtio_pci.ko
-kernel/drivers/net/virtio_net.ko
+kernel/drivers/virtio/virtio_pci.ko*
+kernel/drivers/net/virtio_net.ko*
diff --git a/features.d/usb.modules b/features.d/usb.modules
index d29a9da..39cef3b 100644
--- a/features.d/usb.modules
+++ b/features.d/usb.modules
@@ -1,8 +1,8 @@
 kernel/drivers/usb/host
 kernel/drivers/usb/storage
 kernel/drivers/hid/usbhid
-kernel/drivers/hid/hid-generic.ko
-kernel/drivers/hid/hid-cherry.ko
-kernel/drivers/hid/hid-apple.ko
+kernel/drivers/hid/hid-generic.ko*
+kernel/drivers/hid/hid-cherry.ko*
+kernel/drivers/hid/hid-apple.ko*
 kernel/fs/fat
 kernel/fs/nls
diff --git a/features.d/xenpci.modules b/features.d/xenpci.modules
index 73c212c..4a751fa 100644
--- a/features.d/xenpci.modules
+++ b/features.d/xenpci.modules
@@ -1 +1 @@
-kernel/drivers/pci/xen-pcifront.ko
+kernel/drivers/pci/xen-pcifront.ko*
-- 
2.33.1

